numpy/_core/src/multiarray/abstractdtypes.c

```diff
@@ -276,55 +288,83 @@ complex_common_dtype(PyArray_DTypeMeta *cls, PyArray_DTypeMeta *other)
  *       `Floating`, `Complex`, and `Integer` (both signed and unsigned).
  *       They will have to be renamed and exposed in that capacity.
  */
-NPY_DType_Slots pyintabstractdtype_slots = {
+NPY_NO_EXPORT PyArray_DTypeMeta PyArray_IntAbstractDType = {{{
+        PyVarObject_HEAD_INIT(&PyArrayDTypeMeta_Type, 0)
+        .tp_name = "numpy._IntegerAbstractDType",
+        .tp_base = &PyArrayDescr_Type,
+        .tp_basicsize = sizeof(PyArray_Descr),
+        .tp_flags = Py_TPFLAGS_DEFAULT,
+    },},
+    .type_num = -1,
+    .flags = NPY_DT_ABSTRACT,
+};
+
+NPY_DType_Slots pyintdtype_slots = {
     .discover_descr_from_pyobject = discover_descriptor_from_pyint,
     .default_descr = int_default_descriptor,
     .common_dtype = int_common_dtype,
 };
 
 NPY_NO_EXPORT PyArray_DTypeMeta PyArray_PyIntAbstractDType = {{{
         PyVarObject_HEAD_INIT(&PyArrayDTypeMeta_Type, 0)
-        .tp_name = "numpy._IntegerAbstractDType",
+        .tp_name = "numpy._PythonIntegerDType",
```

Comment:
Side note: if we give a name that looks like it is a place where one can find it, we should actually expose it there....

Reply:
They should also probably live in `np.dtypes` with the other dtype classes.  Right now there's no reason to publicly expose them (I think anyway) but if we ever expose a python interface for the DType API we'll need to do it.